### PR TITLE
[CDEC-453] Add golden/bitripping tests for `HandlerSpec`

### PR DESCRIPTION
**Pull request by Jimbo4350**
## Description

HandlerSpec has partial record accessors which are being removed in: [https://iohk.myjetbrains.com/youtrack/issue/CDEC-403](https://iohk.myjetbrains.com/youtrack/issue/CDEC-403)
Therefore implementing golden/bitripping tests is necessary in order to preserve the correct binary serialization.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CDEC-453

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: infra/cardano-sl-infra.cabal [hunk 1]
```diff
@@ -212,15 +212,41 @@ test-suite test
...
+                    Test.Pos.Infra.Arbitrary
+                    Test.Pos.Infra.Arbitrary.Communication
+                    Test.Pos.Infra.Arbitrary.Slotting
+                    Test.Pos.Infra.Arbitrary.Ssc
+                    Test.Pos.Infra.Arbitrary.Txp
+                    Test.Pos.Infra.Arbitrary.Update
+                    Test.Pos.Infra.Bi
                     Test.Pos.Infra.Diffusion.Subscription.StatusSpec
                     Test.Pos.Infra.Diffusion.Subscription.SubscriptionSpec
+                    Test.Pos.Infra.Gen
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply infra/cardano-sl-infra.cabal]: Yep that is correct. The cabal files weren't properly set up.

File: infra/test/Test/Pos/Infra/Diffusion/Subscription/SubscriptionSpec.hs [hunk 2]
```diff
@@ -6,13 +6,13 @@ import           Control.Concurrent.Async (AsyncCancelled (..), async, cancel,
...
+import           Pos.Infra.Diffusion.Subscription.Common
+                     (SubscriptionTerminationReason (..), networkSubscribeTo)
+import           Prelude
 import           System.IO.Error (userError)
 import           Test.Hspec (Expectation, Spec, describe, expectationFailure,
                      it, shouldBe)
 
-import           Pos.Infra.Diffusion.Subscription.Common
-                     (SubscriptionTerminationReason (..), networkSubscribeTo)
-
```
[the remark this answers is not in the document]
Comment (erikd) [reply infra/test/Test/Pos/Infra/Diffusion/Subscription/SubscriptionSpec.hs]: I think you are Jordan giving credit for something stylish-haskell did :).

File: infra/cardano-sl-infra.cabal [hunk 3]
```diff
@@ -212,15 +212,41 @@ test-suite test
...
                   , hspec
+                  , kademlia
+                  , universum
 
-  ghc-options:      -threaded
+  ghc-options:         -Wall
+                       -O2
+                       -threaded
+
+  default-extensions:  NoImplicitPrelude
```
Comment (intricate) [inline infra/cardano-sl-infra.cabal]: We really shouldn't be adding `default-extensions`.

File: infra/test/Spec.hs [hunk 4]
```diff
@@ -2,6 +2,7 @@ module Spec
     ( spec
     ) where
 
+import           Prelude
```
Comment (intricate) [inline infra/test/Spec.hs]: This should be separated from the below imports.

File: infra/test/Test/Pos/Infra/Bi.hs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+module Test.Pos.Infra.Bi
+       ( tests
+       ) where
+
+import           Universum
+
+import           Hedgehog (Property)
+import qualified Hedgehog as H
+import           Pos.Infra.Communication.Types.Protocol (HandlerSpec (..))
+import qualified Prelude
```
Comment (intricate) [inline infra/test/Test/Pos/Infra/Bi.hs]: This import should be up with `import Universum`

File: infra/cardano-sl-infra.cabal (hunk 3 above)
Comment (mhuesch) [reply infra/cardano-sl-infra.cabal]: I think having `NoImplicitPrelude` is fine, since we seem to use `Universum` everywhere.

**Pull request merged**